OpenCL/m19810_a0-pure.cl

```diff
@@ -0,0 +1,284 @@
+    digests_buf[digests_offset].digest_buf[DGST_R3]
+  };
+
+  /**
+   * base
+   */
+
+  COPY_PW (pws[gid]);
+
+  const u32 salt_len = salt_bufs[salt_pos].salt_len;
+
+  u32 s[64] = { 0 };
+
+  for (int i = 0, idx = 0; i < salt_len; i += 4, idx += 1)
+  {
+    s[idx] = swap32_S (salt_bufs[salt_pos].salt_buf[idx]);
+  }
+
+  /**
+   * loop
+   */
+
+  for (u32 il_pos = 0; il_pos < il_cnt; il_pos++)
+  {
+    pw_t tmp = PASTE_PW;
+
+    tmp.pw_len = apply_rules (rules_buf[il_pos].cmds, tmp.i, tmp.pw_len);
+
+    sha256_ctx_t ctx0;
+
+    sha256_init (&ctx0);
+
+    sha256_update_swap (&ctx0, tmp.i, tmp.pw_len);
+
+    sha256_final (&ctx0);
+
+    const u32 a = ctx0.h[0];
+    const u32 b = ctx0.h[1];
+    const u32 c = ctx0.h[2];
+    const u32 d = ctx0.h[3];
+    const u32 e = ctx0.h[4];
+    const u32 f = ctx0.h[5];
+    const u32 g = ctx0.h[6];
+    const u32 h = ctx0.h[7];
+
+    sha256_ctx_t ctx;
+
+    sha256_init (&ctx);
+
+    ctx.w0[0] = uint_to_hex_lower8_le ((a >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((a >> 24) & 255) << 16;
+    ctx.w0[1] = uint_to_hex_lower8_le ((a >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((a >>  8) & 255) << 16;
+    ctx.w0[2] = uint_to_hex_lower8_le ((b >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((b >> 24) & 255) << 16;
+    ctx.w0[3] = uint_to_hex_lower8_le ((b >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((b >>  8) & 255) << 16;
+    ctx.w1[0] = uint_to_hex_lower8_le ((c >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((c >> 24) & 255) << 16;
+    ctx.w1[1] = uint_to_hex_lower8_le ((c >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((c >>  8) & 255) << 16;
+    ctx.w1[2] = uint_to_hex_lower8_le ((d >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((d >> 24) & 255) << 16;
+    ctx.w1[3] = uint_to_hex_lower8_le ((d >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((d >>  8) & 255) << 16;
+    ctx.w2[0] = uint_to_hex_lower8_le ((e >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((e >> 24) & 255) << 16;
+    ctx.w2[1] = uint_to_hex_lower8_le ((e >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((e >>  8) & 255) << 16;
+    ctx.w2[2] = uint_to_hex_lower8_le ((f >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((f >> 24) & 255) << 16;
+    ctx.w2[3] = uint_to_hex_lower8_le ((f >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((f >>  8) & 255) << 16;
+    ctx.w3[0] = uint_to_hex_lower8_le ((g >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((g >> 24) & 255) << 16;
+    ctx.w3[1] = uint_to_hex_lower8_le ((g >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((g >>  8) & 255) << 16;
+    ctx.w3[2] = uint_to_hex_lower8_le ((h >> 16) & 255) <<  0
+              | uint_to_hex_lower8_le ((h >> 24) & 255) << 16;
+    ctx.w3[3] = uint_to_hex_lower8_le ((h >>  0) & 255) <<  0
+              | uint_to_hex_lower8_le ((h >>  8) & 255) << 16;
+
+    ctx.len = 64;
+
```

Comment:
@jsteube `sha256($pass)` is computed in `ctx0`. What I'm trying to achieve in the above code block is to append the hex values of `ctx0.h` to `s` and compute `sha256` for both of them. Probably there's something wrong in the above block.

Reply:
Please rephrase, I don't understand

Reply:
Nice job